wcfsetup/install/files/lib/core.functions.php

```diff
@@ -90,6 +90,97 @@ function wcfDebug() {
 	}
 }
 
+namespace wcf\functions\deprecated {
+	use \wcf\system\exception\IExtraInformationException;
+
+	function DeprecationError($message, $extraInformation) {
+		// The error needs to be defined lazily, because core.functions.php is initialized before the autoloader is ready.
+		if (!class_exists(DeprecationError::class)) {
+			class DeprecationError extends \Error implements IExtraInformationException {
+				public function __construct($message, $extraInformation) {
+					parent::__construct($message);
+					$this->extraInformation = $extraInformation;
+				}
+				
+				public function getExtraInformation() {
+					return $this->extraInformation;
+				}
+			}
+		}
+		
+		return new DeprecationError($message, $extraInformation);
+	}
+	
+	function feature($message, $extraInformation) {
```

Comment:
Change to `deprecatedFeature`.
Also, when using the function (except for the two functions below), wouldn't you want to pass a string as the second parameter?

Reply:
>  Also, when using the function (except for the two functions below), wouldn't you want to pass a string as the second parameter?

You would use `$message` for a quick message (“`The $baz parameter of Foo::bar() is deprecated`”) and `$extraInformation` for additional details (such as the web link to the docs).